test/CollectionTest_Cpp.cc

```diff
@@ -130,13 +124,9 @@ TEST_CASE_METHOD(CollectionTest_Cpp, "C++ Default Collection", "[Collection]") {
 
 #ifndef NOT_DELETE_DEFAULT_COLLECTION
 TEST_CASE_METHOD(CollectionTest_Cpp, "C++ Delete Default Collection", "[Collection]") {
-    Collection col = db.getDefaultCollection();
-    CHECK(col);
-    
+    CHECK(defaultCollection);
     db.deleteCollection(kCBLDefaultCollectionName);
-    
-    col = db.getDefaultCollection();
-    CHECK(!col);
+    CHECK(!defaultCollection);
```

Comment:
This is not correct because defaultCollection will still hold the original pointer. The original code is correct.